tests/bson/bson-int64-ctor_error-001.phpt

```diff
@@ -0,0 +1,36 @@
+--TEST--
+MongoDB\BSON\Int64 constructor with invalid data
+--FILE--
+<?php
+
+require_once __DIR__ . "/../utils/basic.inc";
+
+$tests = [
+    '0x2', // Hex data
```

Comment:
This makes me wonder how we handle other alternative [integer literals](https://www.php.net/manual/en/language.types.integer.php):

```
<?php
$a = 1234; // decimal number
$a = 0123; // octal number (equivalent to 83 decimal)
$a = 0o123; // octal number (as of PHP 8.1.0)
$a = 0x1A; // hexadecimal number (equivalent to 26 decimal)
$a = 0b11111111; // binary number (equivalent to 255 decimal)
$a = 1_234_567; // decimal number (as of PHP 7.4.0)
```

I won't push for everything, but adding one more test for an octal with leading zeroes seems worthwhile, even if it doesn't result in an failure and ends up in the normal ctor test. It'd be good to have that behavior documented in a test.

Reply:
The alternative notations work fine when used in integer literals as the parser handles this and passes us an integer like any other. However, these values can't be used in strings, so it's not possible to pass octal numbers from the 64-bit range on 32-bit platforms. I don't think this is a particularly popular use case, but it may be worth documenting nonetheless.